app/models/concerns/form526_claim_fast_tracking_concern.rb

```diff
@@ -165,7 +165,79 @@ def get_claim_type
     end
   end
 
+  # Contact the VRO classifier service to classify the contentions on this form
+  # update the form with the classification codes
+  # returns true if all of form's contentions were classified
   def update_classification!
+    if Flipper.enabled?(:disability_526_classifier_multi_contention)
+      update_contention_classification_all!
+    else
+      update_contention_classification_single_contention!
+    end
+  end
+
+  def update_form_with_classification_codes(classified_contentions)
+    classified_contentions.each_with_index do |classified_contention, index|
+      classification_code = classified_contention['classification_code']
+      if classified_contentions[index]['classification_code'].present?
+        form[Form526Submission::FORM_526]['form526']['disabilities'][index]['classificationCode'] = classification_code
+      end
+    end
+
+    update!(form_json: form.to_json)
+    invalidate_form_hash
+  end
+
+  def classify_vagov_contentions(params)
+    vro_client = VirtualRegionalOffice::Client.new
+    response = vro_client.classify_vagov_contentions(params)
+    response.body
+  end
+
+  def format_contention_for_vro(disability)
+    contention = {
+      contention_text: disability['name'],
+      contention_type: disability['disabilityActionType']
+    }
+    contention['diagnostic_code'] = disability['diagnosticCode'] if disability['diagnosticCode']
+    contention
+  end
+
+  # Submits contention information to the VRO contention classification service
+  # adds classification to the form for each contention provided a classification
+  def update_contention_classification_all! # rubocop:disable Metrics/MethodLength
+    contentions_array = disabilities.map { |disability| format_contention_for_vro(disability) }
+    params = {
+      claim_id: saved_claim_id,
+      form526_submission_id: id,
+      contentions: contentions_array
+    }
+    classifier_response = classify_vagov_contentions(params)
+    classifier_response['contentions'].each do |contention|
+      classification = nil
+      if contention.key?(:classification_code) && contention.key?(:classification_name)
+        classification = {
+          classification_code: contention['classification_code'],
+          classification_name: contention['classification_name']
+        }
+      end
+      # NOTE: claim_type is actually type of contention, but formatting
+      # preserved in order to match existing datadog dashboard
+      Rails.logger.info('Classified 526Submission',
+                        id:, saved_claim_id:, classification:,
+                        claim_type: contention['contention_type'])
+    end
+    update_form_with_classification_codes(classifier_response['contentions'])
+    classifier_response['is_fully_classified']
```

Comment:
Is this a new field in the VRO response object? Would it be more robust to just check that the number of classified contentions == the total number of contentions?

Reply:
This is a new field for a new API response. I don't think it would be more robust. This should always be included in the response defined [here](https://github.com/department-of-veterans-affairs/abd-vro/blob/develop/domain-cc/cc-app/src/python_src/pydantic_models.py#L86).